theseus/src/error.rs

```diff
@@ -52,4 +69,35 @@ pub enum Error {
     OtherError(String),
 }
 
+#[derive(Debug)]
+pub struct Error {
```

Comment:
> What's the purpose of this name change? Most Rust libs use `Error`, and I think we should stick to that

@Geometrically - This is the new error type, which wraps `ErrorKind` with the spantrace so that errors can be traced further than the call site.